js/data/bucket.js

```diff
@@ -139,6 +139,21 @@ Bucket.prototype.populateArrays = function() {
 };
 
 /**
+ * Update paint arrays with the given feature properties, leaving geometries
+ * as-is.
+ * @private
+ */
+Bucket.prototype.updatePaintArrays = function(interfaceName, propertiesList) {
```

Comment:
Add a note that the propertiesList is expected to correspond to the source tile's features, _not_ to the bucket's filtered list of features.